library/core/src/intrinsics/mir.rs

```diff
@@ -99,6 +100,30 @@
 //!         Return()
 //!     })
 //! }
+//!
+//! #[custom_mir(dialect = "runtime", phase = "optimized")]
+//! fn push_and_pop<T>(v: &mut Vec<T>, value: T) {
+//!     mir!(
+//!         let unused;
+//!         let popped;
+//!
+//!         {
+//!             Call(unused, pop, Vec::push(v, value))
```

Comment:
I think it would be neat if we actually supported
```
//!             unused = Vec::push(v, value) -> pop
```

Reply:
Yeah, I thought about this. There's a couple reasons I'm hesitant to do this:
 - We'd need to use `=>` instead of `->`, because of follow set rules. That's fine, although it's not the most intuitive syntax
 - More macros means worse error messages
 - Supporting cleanup blocks. Presumably we will eventually want to support those, and I'm not sure what that would look like in connection with this, especially once #102906 lands. Possibly `-> pop, Unreachable()`, `-> pop, Abort()`, or `-> pop, other_block` are fine though?

Reply:
Oh yea, I don't think that syntax should happen in this PR and at best any such syntax should desugar to what you have now.